src/dictionary/open_address_file_hash/open_address_file_hash_dictionary_handler.c

```diff
@@ -182,27 +182,38 @@ oafdict_find(
 			/* copy across the key value as the predicate may be destroyed */
 			memcpy((*cursor)->predicate->statement.range.upper_bound, predicate->statement.range.upper_bound, (((file_hashmap_t *) dictionary->instance)->super.record.key_size));
 
-			/* find the location of the first element as this is a straight equality */
-			int location = cs_invalid_index;
+			oafdict_cursor_t	*oafdict_cursor = (oafdict_cursor_t *) (*cursor);
+			file_hashmap_t		*hash_map		= ((file_hashmap_t *) dictionary->instance);
 
-			/* start at the lowest end of the range and check */
-			if (oafh_find_item_loc((file_hashmap_t *) dictionary->instance, (*cursor)->predicate->statement.range.lower_bound, &location) == err_item_not_found) {
-				/* this will still have to be returned? */
-				(*cursor)->status = cs_end_of_results;
-				return err_ok;
+			(*cursor)->status		= cs_cursor_initialized;
+			oafdict_cursor->first	= (hash_map->map_size) - 1;
+			oafdict_cursor->current = -1;
+
+			err_t err = oafdict_scan(oafdict_cursor);
+
+			if (cs_valid_data != err) {
+				(*cursor)->status = cs_cursor_uninitialized;
```

Comment:
If this signifies that there is no valid data in hashmap, we should return `cs_cursor_end_of_results`